File: algorithm-problems/src/main/java/us/supercheng/algorithm/problems/leetcode/largest3samedigitnumberinstring/Solution.java
package us.supercheng.algorithm.problems.leetcode.largest3samedigitnumberinstring;

class Solution {

    public String largestGoodInteger(String num) {
        Integer ret = null;

        for (int i = 0, right = 0, len = num.length(); i < len; i = right) {
            right = i + 1;
            char c = num.charAt(i);

            while (right < num.length() && c == num.charAt(right))
                right++;

            if (num.charAt(right - 1) == c && right - i > 2) {
                int curr = (c - '0') * 100 + (c - '0') * 10 + (c - '0');
                if (ret == null || ret < curr)
                    ret = curr;
            }
        }

        return ret == null ? "" : ret == 0 ? "000" : String.valueOf(ret);
    }

}

File: algorithm-problems/src/main/java/us/supercheng/algorithm/problems/leetcode/largest3samedigitnumberinstring/Solution.py
class Solution(object):
    def largestGoodInteger(self, num):
        """
        :type num: str
        :rtype: str
        """
        for n in ["999", "888", "777", "666", "555", "444", "333", "222", "111", "000"]:
            if n in num:
                return n

        return ""
